secretstores/hashicorp/vault/vault.go

```diff
@@ -219,7 +219,12 @@ func metadataToTLSConfig(meta *VaultMetadata) *tlsConfig {
 // GetSecret retrieves a secret using a key and returns a map of decrypted string/string values.
 func (v *vaultSecretStore) getSecret(ctx context.Context, secret, version string) (*vaultKVResponse, error) {
 	// Create get secret url
-	vaultSecretPathAddr := fmt.Sprintf("%s/v1/%s/data/%s/%s?version=%s", v.vaultAddress, v.vaultEnginePath, v.vaultKVPrefix, secret, version)
+	var vaultSecretPathAddr string
+	if v.vaultKVPrefix == "" {
+		vaultSecretPathAddr = fmt.Sprintf("%s/v1/%s/data/%s?version=%s", v.vaultAddress, v.vaultEnginePath, secret, version)
+	} else {
+		vaultSecretPathAddr = fmt.Sprintf("%s/v1/%s/data/%s/%s?version=%s", v.vaultAddress, v.vaultEnginePath, v.vaultKVPrefix, secret, version)
```

Comment:
Since we're changing these lines, let's use string concatenation instead of fmt.Sprintf in a hot path, to avoid unnecessary allocations.

```suggestion
		vaultSecretPathAddr = v.vaultAddress + "/v1/" + v.vaultEnginePath + "/data/" + secret + "?version=" + version
```